src/core/include/openvino/op/squeeze.hpp

```diff
@@ -4,22 +4,34 @@
 
 #pragma once
 
+#include <functional>
+#include <utility>
+
 #include "openvino/op/op.hpp"
 
 namespace ov {
 namespace op {
 namespace v0 {
 /// \brief Squeeze operation.
 ///
+/// dimension. The solution mimics the PyTorch approach.
+///
 /// \ingroup ov_ops_cpp_api
 class OPENVINO_API Squeeze : public Op {
 public:
     OPENVINO_OP("Squeeze", "opset1");
 
+    /// \brief Constructs a squeeze operation.
     Squeeze();
-    Squeeze(const Output<Node>& data, const Output<Node>& axes);
-    Squeeze(const Output<Node>& data);
-
+    /// \brief Constructs a squeeze operation.
+    ///
+    /// \param data Input tensor with data
+    /// \param axis The axis along which to squeeze the input tensor.
+    /// \param torch_mode Shape inference result dynamic rank if selected axis has 1 in range of its dynamic
+    Squeeze(const Output<Node>& data, const Output<Node>& axes, const bool torch_mode = false);
+    Squeeze(const Output<Node>& data, const bool torch_mode = false);
```

Comment:
If there is no `axis` there is also no use for `torch_mode` argument, right?

Reply:
Ctor with no axis might still result with dynamic rank, but the reverse infer deduction is not applied. Please share your opinion if this is desired behavior.

https://github.com/openvinotoolkit/openvino/pull/25774/files#diff-23e1a31f71a4d9f2d3a6c63bdb363f5b022b4548fe7419efef1b889bb0b39b4fR99-R101